PowerUp/app/src/main/java/powerup/systers/com/db/DatabaseHandler.java

```diff
@@ -134,10 +137,12 @@ public boolean setSessionId(CharSequence ScenarioName) {
         if (cursor.moveToFirst()) {
             // If the scene is already completed
             if (cursor.getInt(6) == 1) {
+                cursor.close();
                 return false;
             }
             SessionHistory.currSessionID = cursor.getInt(0);
             SessionHistory.currQID = cursor.getInt(5);
+            cursor.close();
             return true;
         }
         cursor.close();
```

Comment:
Instead of adding cursor.close() so many times, what is your opinion if we take a local variable `bool result` and return that after closing the connection just at the end.

Reply:
@anubhakushwaha Can you please explain your suggestion a bit more?